PHP Warnings on SugarWidgetFieldName and SugarWidgetFieldId

#### Issue
From this [thread](https://suitecrm.com/suitecrm/forum/installation-upgrade-help/17848-problem-upgrading-from-7-9-x-to-7-10-0), user sees this in Apache log:

```php

[Wed Feb 21 13:14:21.452511 2018] [:error] [pid 5293] [client 192.168.1.1:51629] PHP Warning: Declaration of SugarWidgetFieldName::displayList(&$layout_def) should be compatible with SugarWidgetField::displayList($layout_def) in /var/www/suitecrm/include/generic/SugarWidgets/SugarWidgetFieldname.php on line 46, referer: https://192.168.0.10/suitecrm/index.php?module=Home&action=index

[Wed Feb 21 13:14:21.452682 2018] [:error] [pid 5293] [client 192.168.1.1:51629] PHP Warning: Declaration of SugarWidgetFieldId::queryFilterIs(&$layout_def) should be compatible with SugarWidgetReportField::queryFilterIs($layout_def) in /var/www/suitecrm/include/generic/SugarWidgets/SugarWidgetFieldid.php on line 51, referer: https://192.168.0.10/suitecrm/index.php?module=Home&action=index
```
And this in `suitecrm.log`:
```
Wed Feb 21 13:14:21 2018 [5293][a0f430d5-13bc-9a57-ebc9-480e1e00bb3c][ERROR] Unable to get proper side for link prospects_301aospects_ida
Wed Feb 21 13:14:21 2018 [5293][a0f430d5-13bc-9a57-ebc9-480e1e00bb3c][ERROR] Unable to get proper side for link prospects_301aospects_ida
Wed Feb 21 13:14:21 2018 [5293][a0f430d5-13bc-9a57-ebc9-480e1e00bb3c][ERROR] Unable to get proper side for link project_name_link
Wed Feb 21 13:14:21 2018 [5293][a0f430d5-13bc-9a57-ebc9-480e1e00bb3c][ERROR] Unable to get proper side for link project_name_link
```

#### Actual Behavior
See the Forum thread, His system seems very broken, I'm not sure this is the only problem, I just know this is worth fixing anyway

#### Environment
* SuiteCRM Version used: 7.9.11
* Environment name and version: PHP 7.0
* Operating System and version: Ubuntu 16.04

--- a/include/generic/SugarWidgets/SugarWidgetFieldenum.php
+++ b/include/generic/SugarWidgets/SugarWidgetFieldenum.php
@@ -88,7 +88,7 @@ class SugarWidgetFieldEnum extends SugarWidgetReportField
 		return $this->_get_column_select($layout_def)." <> ".$this->reporter->db->quoted($input_name0)."\n";
 	}
 
-	public function queryFilterone_of($layout_def) {
+	public function queryFilterone_of($layout_def, $rename_columns = true) {
 		$arr = array ();
 		foreach ($layout_def['input_name0'] as $value) {
 			$arr[] = $this->reporter->db->quoted($value);

--- a/include/generic/SugarWidgets/SugarWidgetFieldid.php
+++ b/include/generic/SugarWidgets/SugarWidgetFieldid.php
@@ -42,7 +42,7 @@ if(!defined('sugarEntry') || !sugarEntry) die('Not A Valid Entry Point');
 class SugarWidgetFieldId extends SugarWidgetReportField
 {
 
- function queryFilterIs(&$layout_def)
+ function queryFilterIs($layout_def)
  {
 		return $this->_get_column_select($layout_def)."='".$GLOBALS['db']->quote($layout_def['input_name0'])."'\n";
  }

--- a/include/generic/SugarWidgets/SugarWidgetFieldmultienum.php
+++ b/include/generic/SugarWidgets/SugarWidgetFieldmultienum.php
@@ -62,7 +62,7 @@ class SugarWidgetFieldMultiEnum extends SugarWidgetFieldEnum {
 		return '('.$query.')';        
 	}
         
-    public function queryFilterone_of(&$layout_def) {
+    public function queryFilterone_of($layout_def, $rename_columns = true) {
         //Fix for inaccurate filtering of contacts in Contacts dashlet on multiselects.
         $arr = array();
         foreach ($layout_def['input_name0'] as $value) {

--- a/include/generic/SugarWidgets/SugarWidgetFieldname.php
+++ b/include/generic/SugarWidgets/SugarWidgetFieldname.php
@@ -67,7 +67,7 @@ class SugarWidgetFieldName extends SugarWidgetFieldVarchar
     }
 
 
-	function displayList(&$layout_def)
+	function displayList($layout_def)
 	{
 		if(empty($layout_def['column_key']))
 		{

--- a/include/generic/SugarWidgets/SugarWidgetFieldsingleenum.php
+++ b/include/generic/SugarWidgets/SugarWidgetFieldsingleenum.php
@@ -43,7 +43,7 @@ if(!defined('sugarEntry') || !sugarEntry) die('Not A Valid Entry Point');
  
 class SugarWidgetFieldSingleEnum extends SugarWidgetFieldEnum {
     
-    function displayInput(&$layout_def) {
+    function displayInput($layout_def) {
         global $app_list_strings;
 
         if(!empty($layout_def['remove_blank']) && $layout_def['remove_blank']) {

--- a/include/generic/SugarWidgets/SugarWidgetFieldtext.php
+++ b/include/generic/SugarWidgets/SugarWidgetFieldtext.php
@@ -62,7 +62,7 @@ class SugarWidgetFieldText extends SugarWidgetFieldVarchar
     }
 
 
-    function queryFilterEquals($layout_def)
+    function queryFilterEquals(&$layout_def)
     {
         return $this->reporter->db->convert($this->_get_column_select($layout_def), "text2char").
         	" = ".$this->reporter->db->quoted($layout_def['input_name0']);

--- a/include/generic/SugarWidgets/SugarWidgetFielduser_name.php
+++ b/include/generic/SugarWidgets/SugarWidgetFielduser_name.php
@@ -43,7 +43,7 @@ if(!defined('sugarEntry') || !sugarEntry) die('Not A Valid Entry Point');
 
 class SugarWidgetFielduser_name extends SugarWidgetFieldname
 {
- function displayInput(&$layout_def) 
+ function displayInput($layout_def) 
  {
         $selected_users = empty($layout_def['input_name0']) ? '' : $layout_def['input_name0'];
  		$str = '<select multiple="true" size="3" name="' . $layout_def['name'] . '[]">' . get_select_options_with_id(get_user_array(false), $selected_users) . '</select>';

--- a/include/generic/SugarWidgets/SugarWidgetFieldvarchar.php
+++ b/include/generic/SugarWidgets/SugarWidgetFieldvarchar.php
@@ -90,7 +90,7 @@ class SugarWidgetFieldVarchar extends SugarWidgetReportField
 		return $this->_get_column_select($layout_def)." LIKE '%".$GLOBALS['db']->quote($layout_def['input_name0'])."'\n";
  }
 
- function queryFilterone_of(&$layout_def)
+ function queryFilterone_of($layout_def, $rename_columns = true)
  {
     foreach($layout_def['input_name0'] as $key => $value) {
         $layout_def['input_name0'][$key] = $GLOBALS['db']->quote($value);

--- a/include/generic/SugarWidgets/SugarWidgetSubPanelConcat.php
+++ b/include/generic/SugarWidgets/SugarWidgetSubPanelConcat.php
@@ -45,7 +45,7 @@ if(!defined('sugarEntry') || !sugarEntry) die('Not A Valid Entry Point');
 
 class SugarWidgetSubPanelConcat extends SugarWidgetField
 {
-	function displayList(&$layout_def)
+	function displayList($layout_def)
 	{
 		$value='';
 		if (isset($layout_def['source']) and is_array($layout_def['source']) and isset($layout_def['fields']) and is_array($layout_def['fields'])) {

--- a/include/generic/SugarWidgets/SugarWidgetSubPanelEditRoleButton.php
+++ b/include/generic/SugarWidgets/SugarWidgetSubPanelEditRoleButton.php
@@ -50,7 +50,7 @@ class SugarWidgetSubPanelEditRoleButton extends SugarWidgetField
 		return '&nbsp;';
 	}
 
-	function displayList(&$layout_def)
+	function displayList($layout_def)
 	{
 		global $app_strings;
         global $subpanel_item_count;

--- a/include/generic/SugarWidgets/SugarWidgetSubPanelEditSecurityGroupUserButton.php
+++ b/include/generic/SugarWidgets/SugarWidgetSubPanelEditSecurityGroupUserButton.php
@@ -53,7 +53,7 @@ class SugarWidgetSubPanelEditSecurityGroupUserButton extends SugarWidgetField
 		return displayList($layout_def);
 	}
 
-	function displayList(&$layout_def)
+	function displayList($layout_def)
 	{
 		global $app_strings;
 		global $image_path;

--- a/include/generic/SugarWidgets/SugarWidgetSubPanelRelFieldEditButton.php
+++ b/include/generic/SugarWidgets/SugarWidgetSubPanelRelFieldEditButton.php
@@ -50,7 +50,7 @@ class SugarWidgetSubPanelRelFieldEditButton extends SugarWidgetField
 		return '&nbsp;';
 	}
 
-	function displayList(&$layout_def)
+	function displayList($layout_def)
 	{
 		die("<pre>" . print_r($layout_def, true) . "</pre>");
 

--- a/include/generic/SugarWidgets/SugarWidgetSubPanelRemoveButtonAccount.php
+++ b/include/generic/SugarWidgets/SugarWidgetSubPanelRemoveButtonAccount.php
@@ -47,7 +47,7 @@ class SugarWidgetSubPanelRemoveButtonAccount extends SugarWidgetSubPanelRemoveBu
 	 * 
 	 * @see SugarWidgetSubPanelRemoveButton::displayList()
 	 */
-	function displayList(&$layout_def) {
+	function displayList($layout_def) {
 		if (!$layout_def['EditView']) {
 			return false;
 		}

--- a/include/generic/SugarWidgets/SugarWidgetSubPanelRemoveButtonProjects.php
+++ b/include/generic/SugarWidgets/SugarWidgetSubPanelRemoveButtonProjects.php
@@ -50,7 +50,7 @@ class SugarWidgetSubPanelRemoveButtonProjects extends SugarWidgetField
 		return '&nbsp;';
 	}
 
-	function displayList(&$layout_def)
+	function displayList($layout_def)
 	{
 		global $app_strings;
 		

--- a/modules/AOW_WorkFlow/aow_utils.php
+++ b/modules/AOW_WorkFlow/aow_utils.php
@@ -637,7 +637,7 @@ function getModuleField(
     $ss->assign("MOD", $mod_strings);
     $ss->assign("APP", $app_strings);
     $ss->assign("module", $module);
-    if ($params['record_id']) {
+    if (!empty($params['record_id'])) {
         $ss->assign("record_id", $params['record_id']);
     }
 

--- a/modules/Calls/Call.php
+++ b/modules/Calls/Call.php
@@ -297,8 +297,9 @@ class Call extends SugarBean {
 	{
 		// First, get the list of IDs.
 		$query = "SELECT contact_id as id from calls_contacts where call_id='$this->id' AND deleted=0";
+		$contact = new Contact();
 
-		return $this->build_related_list($query, new Contact());
+		return $this->build_related_list($query, $contact);
 	}
 
 

--- a/modules/Relationships/RelationshipHandler.php
+++ b/modules/Relationships/RelationshipHandler.php
@@ -231,7 +231,7 @@ class RelationshipHandler extends Relationship {
 
 
 
-	function build_related_list($type="base"){
+	public function build_related_list($type="base", &$template, $row_offset = 0, $limit = -1){
 		//type can be base, rel1
 
 		$target_list = "";

